pandas/core/arrays/array_.py

```diff
@@ -147,6 +147,12 @@ def array(data,         # type: Sequence[object]
         period_array, ExtensionArray, IntervalArray
     )
 
+    if lib.is_scalar(data):
+        msg = (
+            "Cannot pass scalar '{}' to 'pandas.array'."
+        )
+        raise ValueError(msg.format(data))
```

Comment:
You could also combine this check with checking in the end that the returned value has a dimension of 1 (so not 0 for scalars or >= 2)

Reply:
I think we should have it early, to avoid passing scalars to `ExtensionArray._from_sequence`. Subclasses should probably validate that on their own, but I think we'd like a consistent error message.